spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,131 @@
+      expect(page).to have_content("Price: $#{@item1.price}")
+      expect(page).to have_content("Supermarket Name: #{@supermarket1.name}")
+    end
+
+    within("#item-#{@item2.id}") do
+      expect(page).to have_content("Item: #{@item2.name}")
+      expect(page).to have_content("Price: $#{@item2.price}")
+      expect(page).to have_content("Supermarket Name: #{@supermarket1.name}")
+    end
+
+    within("#item-#{@item3.id}") do
+      expect(page).to have_content("Item: #{@item3.name}")
+      expect(page).to have_content("Price: $#{@item3.price}")
+      expect(page).to have_content("Supermarket Name: #{@supermarket1.name}")
+    end
+
+    expect(page).to_not have_content(@customer2.name)
+
+    expect(page).to_not have_content(@item5.name)
+    expect(page).to_not have_content(@item5.price)
+    expect(page).to_not have_content(@item6.name)
+    expect(page).to_not have_content(@item6.price)
+    expect(page).to_not have_content(@item7.name)
+    expect(page).to_not have_content(@item7.price)
+
+    expect(page).to_not have_content(@supermarket2.name)
+  end
+
+  it "displays another customer's name and a list of their items and all their attributes" do 
+    visit "/customers/#{@customer2.id}"
+
+    expect(page).to have_content(@customer2.name)
+
+    within("#item-#{@item5.id}") do
+      expect(page).to have_content("Item: #{@item5.name}")
+      expect(page).to have_content("Price: $#{@item5.price}")
+      expect(page).to have_content("Supermarket Name: #{@supermarket2.name}")
+    end
+
+    within("#item-#{@item6.id}") do
+      expect(page).to have_content("Item: #{@item6.name}")
+      expect(page).to have_content("Price: $#{@item6.price}")
+      expect(page).to have_content("Supermarket Name: #{@supermarket2.name}")
+    end
+
+    within("#item-#{@item7.id}") do
+      expect(page).to have_content("Item: #{@item7.name}")
+      expect(page).to have_content("Price: $#{@item7.price}")
+      expect(page).to have_content("Supermarket Name: #{@supermarket2.name}")
+    end
+
+    expect(page).to_not have_content(@customer1.name)
+
+    expect(page).to_not have_content(@item1.name)
+    expect(page).to_not have_content(@item1.price)
+    expect(page).to_not have_content(@item2.name)
+    expect(page).to_not have_content(@item2.price)
+    expect(page).to_not have_content(@item3.name)
+    expect(page).to_not have_content(@item3.price)
+
+    expect(page).to_not have_content(@supermarket1.name)
+  end
+
+  # User Story 2
+  it "can fill out a form to add an item to this customer" do
+    visit "/customers/#{@customer1.id}"
+
+    fill_in :item_id, with: @item8.id
+    click_button "Submit"
+
+    expect(current_path).to eq("/customers/#{@customer1.id}")
+
+    within("#item-#{@item8.id}") do
+      expect(page).to have_content("Item: #{@item8.name}")
+      expect(page).to have_content("Price: $#{@item8.price}")
+      expect(page).to have_content("Supermarket Name: #{@supermarket2.name}")
+    end
```

Comment:
What would make this test even better is if first in this test, you checked the page to ensure item8.name was NOT on the page, then did the form actions then tested that item8 was on the page. This will show the full scope of expected behavior for this feature.